File: LeetCode/Problems/00007. Reverse Integer/Solution.py

class Solution:
    def reverse(self, x: int) -> int:
        if ( x >= 2**31 or x < -2**31 ):
            return 0
        else: 
            sign = -1 if x < 0 else 1
            x *= sign
            ans = 0

            while x:
                ans = ans*10 + (x%10)
                x //= 10

            ans *= sign
            return ans


if __name__ == '__main__':
    input_x = 123
    
    solution = Solution()
    output = solution.reverse(input_x)
    print("Output: ", output)

File: LeetCode/Problems/00007. Reverse Integer/Solution.cpp
#include <iostream>
#include <cmath>
using namespace std;

class Solution {
public:
    int reverse(int x) {
        if ( x >= pow(2,31) || x < -pow(2,31) ){
            return 0;
        }else{
            int sign = (x < 0)?-1:1;
            int ans =0;
            while( x ){
                ans = ans * 10 + (x % 10);
                x /= 10;
            }
            ans *= sign;
            return ans;
        }
        
    }
};

int main() {
    int input_x = 123;
    int output;

    Solution solution;
    output = solution.reverse(input_x);


    cout << "output: " << output << endl;
    
    return 0;
}



